astropy/nddata/nduncertainty.py

```diff
@@ -13,146 +13,54 @@
-    does not support correlated uncertainties.
-
-Returns
--------
-result_uncertainty : {instance} instance
-    The resulting uncertainty
-
-Raises
-------
-ValueError
-    Raised if the uncertainty arrays or resulting data cannot be used together
-    for the arithmetic operation (not broadcastable) or conflicting units.
-
-Notes
------
-Handling units (especially if the differ from the parents unit) is done in here
-but be aware that since there are no checks for the unit, incorrect
-assigned units may break the uncertainty propagation even if the resulting data
-(with units) can be computed.
-"""
-
-
 class IncompatibleUncertaintiesException(Exception):
-    """
-    This exception should be used to indicate cases in which uncertainties
+    """This exception should be used to indicate cases in which uncertainties
     with two different classes can not be propagated.
     """
 
 
 class MissingDataAssociationException(Exception):
-    """
-    This exception should be used to indicate that an uncertainty instance has
-    not been associated with a parent `~astropy.nddata.NDData` object.
+    """This exception should be used to indicate that an uncertainty instance
+    has not been associated with a parent `~astropy.nddata.NDData` object.
     """
 
 
 @six.add_metaclass(ABCMeta)
 class NDUncertainty(object):
-    """
-    This is the base class for uncertainty classes used with NDData.
+    """This is the metaclass for uncertainty classes used with `NDData`.
 
     Parameters
     ----------
-    array: any type, optional
+    array : any type, optional
         The array or value (the parameter name is due to historical reasons) of
-        the uncertainty.
-        `~numpy.ndarray`, `~astropy.units.Quantity` or `NDUncertainty`
-        subclasses are recommended.
-        If the `array` is `list`-like or `~numpy.ndarray`-like it will be cast
-        to a base `~numpy.ndarray`.
+        the uncertainty. `numpy.ndarray`, `~astropy.units.Quantity` or
+        `NDUncertainty` subclasses are recommended.
+        If the `array` is `list`-like or `numpy.ndarray`-like it will be cast
+        to a plain `numpy.ndarray`.
         Default is ``None``.
 
-    unit: `~astropy.units.Unit` or str, optional
-        The unit of the uncertainty ``array``. If input is a string this will
-        be converted to an `~astropy.units.Unit`.
+    unit : `~astropy.units.Unit` or str, optional
+        Unit for the uncertainty ``array``. Strings that can be converted to a
+        `~astropy.units.Unit` are allowed.
         Default is ``None``.
 
     copy : `bool`, optional
-        Save the array as copy or as reference. ``True`` copies the
-        uncertainty array before saving it while ``False`` tries to save it
-        as reference. Note however that it is not always possible to
-        save is as reference.
+        Save the ``array`` as copy? ``True`` copies it before saving, while
```

Comment:
Ah, now I understand what you mean by the question mark. I find the phrasing the confusing though (I don't expect the description to start with a question). How about: "Indicates whether to save the ``array`` as a copy." instead?